// @ant-design/compatible/lib/theme/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "darkAlgorithm", {
  enumerable: true,
  get: function get() {
    return _dark.default;
  }
});
exports.darkTheme = void 0;
Object.defineProperty(exports, "defaultAlgorithm", {
  enumerable: true,
  get: function get() {
    return _default.default;
  }
});
exports.defaultTheme = void 0;

var _default = _interopRequireDefault(require("./default"));

var _dark = _interopRequireDefault(require("./dark"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var v4Token = {
  token: {
    borderRadius: 2,
    colorPrimary: '#1890ff',
    wireframe: true
  }
};

var defaultTheme = _objectSpread(_objectSpread({}, v4Token), {}, {
  algorithm: _default.default,
  components: {
    Menu: {
      radiusItem: 0,
      radiusSubMenuItem: 0,
      colorItemTextHover: '#1890ff',
      colorItemTextSelected: '#1890ff',
      colorItemBgSelected: '#e6f7ff',
      colorActiveBarWidth: 3,
      itemMarginInline: 0,
      colorItemBgHover: 'transparent'
    }
  }
});

exports.defaultTheme = defaultTheme;

var darkTheme = _objectSpread(_objectSpread({}, v4Token), {}, {
  algorithm: _dark.default,
  components: {
    Menu: {
      radiusItem: 0,
      radiusSubMenuItem: 0,
      colorItemTextHover: '#1890ff',
      colorItemTextSelected: '#1890ff',
      colorItemBgSelected: '#111b26',
      colorActiveBarWidth: 3,
      itemMarginInline: 0,
      colorItemBgHover: 'transparent'
    }
  }
});

exports.darkTheme = darkTheme;
